File: problems/938/solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def rangeSumBST(self, root: TreeNode, L: int, R: int) -> int:
        val = 0
        
        if root:
            if root.val >= L and root.val <= R:
                val += root.val
        
            if root.left:
                val += self.rangeSumBST(root.left, L, R)
        
            if root.right:
                val += self.rangeSumBST(root.right, L, R)
        
        return val

File: problems/938/solution.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int rangeSumBST(TreeNode* root, int low, int high) {
        if (not root) {
            return 0;
        }
        
        int ans = 0;
        
        if (root->val >= low and root->val <= high) {
            ans += root->val;
        }
        
        if (root->val > low) {
            ans += rangeSumBST(root->left, low, high);
        }
        
        if (root->val < high) {
            ans += rangeSumBST(root->right, low, high);
        }
        
        return ans;
    }
};